front-end/src/Pages/Client/Client-Product/index.jsx

```diff
@@ -4,23 +4,24 @@ import { useSelector, useDispatch } from 'react-redux';
 import { getApiData } from '../../../Redux/action/apiProductsAction';
 import ProductCard from './innerPage/ProductCard';
 import convertBRL from '../../../Services/BRLFunction';
+import { shoppingListAction } from '../../../Redux/action/shoppingListAction';
 import './styles.css';
 
 const ClientProduct = () => {
   const products = useSelector(state => state.apiProductsReducer.data);
   const shoppingList = useSelector(state => state.shoppingListReducer.data);
   const dispatch = useDispatch();
-  
+
   const totalPrice = shoppingList.reduce((acc, { price, sellingQnt }) => acc + price * sellingQnt, 0);
 
   const totalPriceBRL = convertBRL(totalPrice)
 
-  const goCart = () => {
-    localStorage.setItem('sellingProducts', JSON.stringify(shoppingList));
-  };
-
   useEffect(() => {
     dispatch(getApiData());
+    if (localStorage.getItem('sellingProducts')) {
+      const shoppingListLocalStorage = JSON.parse(localStorage.getItem('sellingProducts'));
+      dispatch(shoppingListAction(shoppingListLocalStorage));
+    }
```

Comment:
Assim não ficaria melhor??

```suggestion
      const shoppingListLocalStorage = JSON.parse(localStorage.getItem('sellingProducts'));
      shoppingListLocalStorage && dispatch(shoppingListAction(shoppingListLocalStorage));
    }
```

Reply:
gostei da sugestão irei altera-lo